src/Parser/KeyParser.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the ACME PHP library.
+ *
+ * (c) Titouan Galopin <galopintitouan@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace AcmePhp\Ssl\Parser;
+
+use AcmePhp\Ssl\Exception\KeyParsingException;
+use AcmePhp\Ssl\Key;
+use AcmePhp\Ssl\ParsedKey;
+
+/**
+ * Parse keys to extract metadata.
+ *
+ * @author Titouan Galopin
+ */
+class KeyParser
+{
+    /**
+     * Parse the key.
+     *
+     * @param Key $key
+     *
+     * @return ParsedKey
+     */
+    public function parse(Key $key)
+    {
+        if (!$key->getResource()) {
+            throw new KeyParsingException(sprintf('Fail to parse key with error: %s', openssl_error_string()));
```

Comment:
this function should not be aware that getResource use opessl .`openssl_error_string` should not be used here